### PR TITLE
aws functional tests documentation

### PR DESCRIPTION
**Pull request by gurevichdmitry**
### Summary of your changes

Updating testing docs - adding guidelines how to develop AWS CIS functional tests.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+
+Test file defines the logic of the test, and the data defines test cases permutation.
+
+Tests data location: `<project_root>/tests/product/tests/data/aws`
+
+Data file identification: `aws_`
+
+For example `EC2` data cases will be located under `aws_ec2_test_cases.py`.
+
+##### Adding new CSPM test
```
Comment (oren-zohar) [inline tests/README.md:178]:
```suggestion
#### Adding new CSPM test
```

File: tests/README.md [hunk 2]
```diff
@@ -146,6 +146,92 @@ Additional functionality
 - **caplog fixture** - add a sink that propagates Loguru to the caplog handler.
 - **logger_wraps** - useful to log entry and exit values of a function
 
+#### CSPM Functional Tests
```
Comment (oren-zohar) [inline tests/README.md:149]:
```suggestion
### CSPM Functional Tests
```

File: tests/README.md [hunk 3]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+Data file identification: `aws_`
+
+For example `EC2` data cases will be located under `aws_ec2_test_cases.py`.
+
+##### Adding new CSPM test
+
+- Define data manually in AWS Cloud and define / get property for resource unique identification
+- Create test case data in `data` folder, for example in file `aws_logging_test_cases.py`
+```
+cis_aws_log_3_1_pass = EksAwsServiceCase(
```
Comment (oren-zohar) [inline tests/README.md:183]:
```suggestion
```python
cis_aws_log_3_1_pass = EksAwsServiceCase(
```

File: tests/README.md [hunk 4]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+    rule_tag=CIS_3_1,
+    case_identifier="cloudtrail-704479110758", # resource unique identifier
+    expected=RULE_PASS_STATUS,
+)
+```
+- Update test cases dictionary or create new if not exist, for example
```
Comment (oren-zohar) [inline tests/README.md:189]:
```suggestion
- Update the test cases dictionary or create a new one if it didn't exist, for example:
```

File: tests/README.md [hunk 5]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+```
```
Comment (oren-zohar) [inline tests/README.md:190]:
```suggestion
```python
```

File: tests/README.md [hunk 6]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+cis_aws_log_3_1 = {
+    "3.1 Ensure CloudTrail is enabled in all regions expect: passed": cis_aws_log_3_1_pass,
+}
+```
+- Finally, reference created dictionary in the group of all test cases, for example
+```
```
Comment (oren-zohar) [inline tests/README.md:196]:
```suggestion
```python
```

File: tests/README.md [hunk 7]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+cis_aws_log_cases = {
+    **cis_aws_log_3_1,
+    ...
+```
+- If just adding new test case to exist test suite no additional steps required, the case will be added automatically
```
Comment (oren-zohar) [inline tests/README.md:201]:
```suggestion
- If just adding a new test case to existing test suite, no additional steps are required, the case will be added automatically
```

File: tests/README.md [hunk 8]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+- For new test suite create a test file in `./product/tests` folder, like `test_aws_logging_rules.py`
```
Comment (oren-zohar) [inline tests/README.md:202]: we can link here too

File: tests/README.md [hunk 9]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+- Implement test method or just copy from any `test_aws_` and updated accordingly data section
+```
```
Comment (oren-zohar) [inline tests/README.md:204]:
```suggestion
```python
```

File: tests/README.md [hunk 10]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+    test_aws_logging_rules, # should be updated
+    Parameters(
+        ("rule_tag", "case_identifier", "expected"),
+        [*aws_logging_tc.cis_aws_log_cases.values()], # should be replaced by new data
+        ids=[*aws_logging_tc.cis_aws_log_cases.keys()], # should be replaced by new data
+    ),
+)
+```
+- Define new marker, for example
+```
```
Comment (oren-zohar) [inline tests/README.md:215]:
```suggestion
```python
```

File: tests/README.md [hunk 11]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+@pytest.mark.aws_logging_rules # <-- new marker should be created
+def test_aws_logging_rules(
+    elastic_client,
+    cloudbeat_agent,
+```
+
+- Update markers section in pyproject.toml with newly created marker
+```
```
Comment (oren-zohar) [inline tests/README.md:223]:
```suggestion
```python
```

File: tests/README.md [hunk 12]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+- Define new marker, for example
+```
+@pytest.mark.aws_logging_rules # <-- new marker should be created
+def test_aws_logging_rules(
+    elastic_client,
+    cloudbeat_agent,
+```
+
+- Update markers section in pyproject.toml with newly created marker
```
Comment (oren-zohar) [inline tests/README.md:222]:
```suggestion
- Update markers section in `pyproject.toml` with newly created marker
```

File: tests/README.md [hunk 13]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+```
+[tool.pytest.ini_options]
+markers = [
+    "pre_merge",
+    "pre_merge_agent",
+    ... # <-- add new marker
+```
+
+- Execute test suite by running the following command and replacing marker `aws_logging_rules` by newly defined marker
```
Comment (oren-zohar) [inline tests/README.md:231]:
```suggestion
- Execute the test suite by running the following command and replacing marker `aws_logging_rules` with newly defined marker
```

File: tests/README.md [hunk 14]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+```
```
Comment (oren-zohar) [inline tests/README.md:232]:
```suggestion
```shell
```

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: tests/README.md [hunk 15]
```diff
@@ -146,6 +146,92 @@ Additional functionality
 - **caplog fixture** - add a sink that propagates Loguru to the caplog handler.
 - **logger_wraps** - useful to log entry and exit values of a function
 
+#### CSPM Functional Tests
+
+CSPM functional tests are designed to check CSPM rules behaviour.
+
+Tests folder location: `<project_root>/tests/product/tests`
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/README.md]: provided relative link

File: tests/README.md [hunk 16]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+- Identification And Management (IAM)
+- Logging
+- Monitoring
+- Relation Database Service (RDS)
+- Simple Storage Service (S3)
+- Networking (VPC)
+
+For example `EC2` rules will be located under `test_aws_ec2_rules.py`
+
+In order to separate and simply test development process, parametrization decorator is used.
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/README.md]: I rephrased the sentence

File: tests/README.md [hunk 17]
```diff
@@ -146,6 +146,92 @@ Additional functionality
...
+
+Test file defines the logic of the test, and the data defines test cases permutation.
+
+Tests data location: `<project_root>/tests/product/tests/data/aws`
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/README.md]: relative link is added.

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**